what4/src/What4/Interface.hs

```diff
@@ -312,6 +312,9 @@ class HasAbsValue e => IsExpr e where
 
   asAffineVar :: e tp -> Maybe (ConcreteVal tp, e tp, ConcreteVal tp)
 
+  -- | Does the expression reference a Bound Variable (@BoundVar@)?
+  isBoundVar :: e tp -> Bool
```

Comment:
What about `asBoundVar :: e tp -> Maybe (BoundVar tp)` ?

I think that is a potentially useful operation too.  Then `isBoundVar = isJust . asBoundVar`

Reply:
Sure, that sounds reasonable.

Reply:
Update: we determined this isn't possible because it's a `BoundVar sym tp`.